### PR TITLE
Fix dryrun PE Task related bugs

### LINKED ISSUE #245: [PE-D] add command for Events and Deadlines shows incorrect error message
### Expected Behavior

When the commands `add -e` or `add -d` are run without a description, an error message should indicate that no description has been provided.

### Actual Behavior

When the commands `add -e` or `add -d` are run without a description, they show the error message that the date and time are missing even when they are provided.

## Steps to Reproduce:

1. Execute `add -e d/2020-01-01 t/00:00` in the command box.

1. The following output is produced:

1. Notice that the error message asks to `include task DATE and TIME d/YYYY-MM-DD t/HH:MM` even when valid date and time have been provided.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: w-yeehong/ped#1

### LINKED ISSUE #255: [PE-D] Invalid date is stored unexpectedly
Build Number: Version Number 1.3
Reported By: Chu Yi-Ting
Reported On: 2020-10-30
Reason: Defect
Severity: Low
Environment: macOS version 10.15.7
Description:
Create a task with date 2019-02-29, a task with date 2019-02-28 is created.

Expected Result: Task with date 2019-03-01 or invalid date error response
Actual Result: The application stuck and no response is provided

Steps To Reproduce:
1.	Type: add -e my todo task d/2019-02-29 t/12:00

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: chuyiting/ped#5

### LINKED ISSUE #256: [PE-D] Error For creating a deadline event
Build Number: Version Number 1.3
Reported By: Chu Yi-Ting
Reported On: 2020-10-30
Reason: Defect
Severity: High
Environment: macOS version 10.15.7

Description:
Cannot create deadline events

Steps To Reproduce:
1.	Type: add -d CS2103T Week 5 ip tasks d/2020-09-08 t/23:59
2.	Application stuck and no response is provided

-------------
Labels: `severity.High` `type.FunctionalityBug`
original: chuyiting/ped#6

### LINKED ISSUE #257: [PE-D] Unnatural index for task
Reported By: Chu Yi-Ting
Reported On: 2020-10-30
Reason: Defect
Severity: Very Low
Environment: macOS version 10.15.7

Description:
The first Deadline Task being added is marked as D5 if it is created as the fifth task, which might be unintuitive for the user

Expected Behavior: 
The first Deadline Task is marked as D1 or the task id is marked as id [D] and the deletion is based on the “task” index.

Actual Behavior: 
The index of the first deadline is D(nth-task).

Steps To Reproduce:
1.	Add -t my first task
2.	Add -d my first deadline d/2020-10-30 t/16:00

Suggestions:
1.	Change the output to something similar 1[D] is more intuitivefor me   
2.	Alternatively, changing the implementation to show D1 for the first deadline is also a good idea to me.

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: chuyiting/ped#7

### LINKED ISSUE #276: Clear task related bugs found from PE dry run

### PR DESCRIPTION
**Pull request by Asuraxsoul**
Bug fixes: Fixes #276 
Fixes #245 
Fixes #255 
Fixes #256 
Add more specific description and explanation for Task ID in UG
Fixes #257

### CODE REVIEW

File: src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java [hunk 1]
```diff
@@ -45,14 +45,14 @@ public static Event parseEvent(String[] nameKeywords, int length) throws ParseEx
         LocalDateTime formattedEventDateTime;
 
         try {
+            String eventDescription = parseTimedTaskDescription(nameKeywords, length);
             formattedEventDateTime = parseTimedTaskTime(nameKeywords, length);
+
+            return new Event(eventDescription, formattedEventDateTime);
+
         } catch (ParseException pe) {
```
Comment (wilinetan) [inline src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java:53]: Will it be better to have a more helpful exception message thrown?

File: src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java [hunk 2]
```diff
@@ -63,26 +63,37 @@ public static Deadline parseDeadline(String[] nameKeywords, int length) throws P
...
+        }
+
         for (int i = 2; i < length; i++) {
-            if (nameKeywords[i].substring(0, 2).equals(TASK_DATE)) {
+            if (nameKeywords[i].length() <= 1) {
+                //Ignores the string segment if the length is <= 1
+
+            } else if (nameKeywords[i].substring(0, 2).equals(TASK_DATE)) {
                 datePrefixLocation = i;
             }
```
Comment (wilinetan) [inline src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java:98]: to shorten the code for this part, perhaps you can consider just checking for the second condition

File: src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java [hunk 3]
```diff
@@ -109,12 +121,14 @@ public static LocalDateTime parseTimedTaskTime(String[] nameKeywords, int length
         int timePrefixLocation = -1;
 
         for (int i = 2; i < length; i++) {
-            if (nameKeywords[i].substring(0, 2).equals(TASK_DATE)) {
+            if (nameKeywords[i].length() <= 1) {
+                //Ignores the string segment if the length is <= 1
+
+            } else if (nameKeywords[i].substring(0, 2).equals(TASK_DATE)) {
```
Comment (wilinetan) [inline src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java:127]: same comment as above on shortening code

**Review by wilinetan — COMMENTED** [review]
Thanks for fixing the bugs!

File: src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java (hunk 2 above)
Comment (Asuraxsoul) [reply src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java:98]: yes I understand where you are coming from but the first if condition is used to solve bug #256

File: src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java (hunk 3 above)
Comment (Asuraxsoul) [reply src/main/java/seedu/jarvis/logic/parser/TaskCommandParser.java:127]: same comment as the above

**Comment by Asuraxsoul** [comment]
> Thanks for fixing the bugs!

Thanks @wilinetan

**Pull request merged**